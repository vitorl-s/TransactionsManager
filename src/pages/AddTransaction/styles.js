import {StyleSheet, Platform} from 'react-native';
import {Colors} from '../../constants/Colors';

const styles = StyleSheet.create({
  container: {
    flex: 1,
    paddingTop: 10,
    paddingBottom: 20,
    backgroundColor: Colors.Background,
  },
  inputContainer: {
    width: '90%',
    flex: 1,
    justifyContent: 'center',
    alignSelf: 'center',
  },
  inputLabel: {
    color: Colors.Label,
    fontSize: 18,
    marginTop: 10,
    fontWeight: 'bold',
  },
  input: {
    color: Colors.Text,
    marginLeft: 5,
    padding: 15,
  },
  button: {
    backgroundColor: Colors.ButtonColor,
    width: '90%',
    justifyContent: 'center',
    alignItems: 'center',
    alignSelf: 'center',
    padding: 15,
    borderRadius: 25,
  },
  buttonText: {
    color: Colors.Text,
    fontSize: 16,
    fontWeight: 'bold',
  },
  datepicker: {
    width: 115,
  },
  dateInput: {
    borderWidth: 0,
  },
  dateText: {
    color: Colors.Text,
  },
});

export default styles;
