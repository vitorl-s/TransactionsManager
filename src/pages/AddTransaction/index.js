import React, {useState} from 'react';
import moment from 'moment';
import DatePicker from 'react-native-datepicker';
import {View, Text, TextInput, TouchableOpacity, Alert} from 'react-native';
import {useDispatch} from 'react-redux';
import {useNavigation} from '@react-navigation/native';
import styles from './styles';
import {Colors} from '../../constants/Colors';
import {SaveTransaction} from '../../redux/actions/transaction';

const AddTransaction = () => {
  const [value, setValue] = useState();
  const [desc, setDesc] = useState();
  const parsedDate = moment().format('DD/MM/YYYY');
  const [date, setDate] = useState(parsedDate);
  const navigation = useNavigation();
  const dispatch = useDispatch();

  const onChange = (selectedDate) => {
    const currentDate = selectedDate || date;
    setDate(moment(currentDate, 'DD-MM-YYYY').format('DD/MM/YYYY'));
  };

  const handleButtonPress = () => {
    if (date && desc && value) {
      const formatedValue = parseFloat(value);
      const transaction = {
        date,
        value: formatedValue,
        desc,
      };
      dispatch(SaveTransaction(transaction));
      navigation.navigate('Home');
    } else {
      Alert.alert('Preencha todos os campos para criar sua transação');
    }
  };

  return (
    <View style={styles.container}>
      <View style={styles.inputContainer}>
        <Text style={styles.inputLabel}>Valor da transação(R$)</Text>
        <TextInput
          value={value}
          placeholder="Valor"
          placeholderTextColor={Colors.Text}
          style={styles.input}
          keyboardType="decimal-pad"
          underlineColorAndroid={Colors.Text}
          onChangeText={(price) => setValue(price)}
        />
        <Text style={styles.inputLabel}>Descrição</Text>
        <TextInput
          value={desc}
          placeholder="Descrição"
          placeholderTextColor={Colors.Text}
          underlineColorAndroid={Colors.Text}
          style={styles.input}
          onChangeText={(description) => setDesc(description)}
        />
        <Text style={styles.inputLabel}>Data</Text>
        <TouchableOpacity>
          <DatePicker
            date={date}
            style={styles.datepicker}
            mode="date"
            androidMode="spinner"
            showIcon={false}
            format="DD/MM/YYYY"
            maxDate="31/12/2020"
            confirmBtnText="Confirmar"
            cancelBtnText="Cancelar"
            customStyles={{
              dateInput: styles.dateInput,
              dateText: styles.dateText,
            }}
            onDateChange={onChange}
          />
        </TouchableOpacity>
      </View>
      <TouchableOpacity style={styles.button} onPress={handleButtonPress}>
        <Text style={styles.buttonText}>Adicionar Transação</Text>
      </TouchableOpacity>
    </View>
  );
};

export default AddTransaction;
