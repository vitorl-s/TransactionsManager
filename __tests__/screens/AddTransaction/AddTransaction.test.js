import React from 'react';
import configureStore from 'redux-mock-store';
import {Provider} from 'react-redux';
import {render, fireEvent} from '@testing-library/react-native';
import AddTransaction from '../../../src/pages/AddTransaction/index';
import MockedNavigator from '../../../src/navigator/mockedNavigator';

describe('AddTransactionPage tests', () => {
  test('should render AddTransaction Page', () => {
    const initialState = {
      transactions: {
        transactions: [
          {value: 40.5, desc: 'testing description', date: '08/08/2020'},
        ],
      },
    };
    const mockStore = configureStore();
    const store = mockStore(initialState);
    const {toJSON} = render(
      <Provider store={store}>
        <MockedNavigator component={AddTransaction} />
      </Provider>,
    );
    expect(toJSON()).toMatchSnapshot();
  });

  test('should fill addTransaction inputs', () => {
    const initialState = {
      transactions: {
        transactions: [
          {value: 40.5, desc: 'testing description', date: '08/08/2020'},
        ],
      },
    };
    const mockStore = configureStore();
    const store = mockStore(initialState);
    const {getByPlaceholderText} = render(
      <Provider store={store}>
        <MockedNavigator component={AddTransaction} />
      </Provider>,
    );
    const valueInput = getByPlaceholderText('Valor');
    fireEvent.changeText(valueInput, '12');
    const descriptionInput = getByPlaceholderText('Descrição');
    fireEvent.changeText(descriptionInput, 'This is a description test');
    expect(valueInput.props.value).toEqual('12');
    expect(descriptionInput.props.value).toEqual('This is a description test');
  });
});
