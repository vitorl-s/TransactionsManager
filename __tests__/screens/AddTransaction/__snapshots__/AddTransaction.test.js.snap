// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddTransactionPage tests should render AddTransaction Page 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RNCSafeAreaProvider
    onInsetsChange={[Function]}
    style={
      Array [
        Object {
          "flex": 1,
        },
        undefined,
      ]
    }
  >
    <View
      pointerEvents="box-none"
      style={
        Array [
          Object {
            "zIndex": 1,
          },
          false,
        ]
      }
    >
      <View
        accessibilityElementsHidden={false}
        importantForAccessibility="auto"
        onLayout={[Function]}
        pointerEvents="box-none"
        style={null}
      >
        <View
          pointerEvents="box-none"
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "opacity": 1,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "zIndex": 0,
            }
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "rgb(255, 255, 255)",
                "borderBottomColor": "rgb(216, 216, 216)",
                "flex": 1,
                "shadowColor": "rgb(216, 216, 216)",
                "shadowOffset": Object {
                  "height": 0.5,
                  "width": 0,
                },
                "shadowOpacity": 0.85,
                "shadowRadius": 0,
              }
            }
          />
        </View>
        <View
          pointerEvents="box-none"
          style={
            Object {
              "height": 64,
              "maxHeight": undefined,
              "minHeight": undefined,
              "opacity": undefined,
              "transform": undefined,
            }
          }
        >
          <View
            pointerEvents="none"
            style={
              Object {
                "height": 20,
              }
            }
          />
          <View
            pointerEvents="box-none"
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "flexDirection": "row",
                "justifyContent": "center",
              }
            }
          >
            <View
              pointerEvents="box-none"
              style={
                Object {
                  "marginHorizontal": 16,
                  "opacity": 1,
                }
              }
            >
              <Text
                accessibilityRole="header"
                numberOfLines={1}
                onLayout={[Function]}
                style={
                  Object {
                    "color": "rgb(28, 28, 30)",
                    "fontSize": 17,
                    "fontWeight": "600",
                  }
                }
              >
                MockedScreen
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View
        pointerEvents="box-none"
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <View
          collapsable={false}
          style={
            Object {
              "opacity": 1,
            }
          }
        />
        <View
          accessibilityElementsHidden={false}
          closing={false}
          gestureVelocityImpact={0.3}
          importantForAccessibility="auto"
          onClose={[Function]}
          onGestureBegin={[Function]}
          onGestureCanceled={[Function]}
          onGestureEnd={[Function]}
          onOpen={[Function]}
          onTransitionStart={[Function]}
          pointerEvents="box-none"
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
          transitionSpec={
            Object {
              "close": Object {
                "animation": "spring",
                "config": Object {
                  "damping": 500,
                  "mass": 3,
                  "overshootClamping": true,
                  "restDisplacementThreshold": 10,
                  "restSpeedThreshold": 10,
                  "stiffness": 1000,
                },
              },
              "open": Object {
                "animation": "spring",
                "config": Object {
                  "damping": 500,
                  "mass": 3,
                  "overshootClamping": true,
                  "restDisplacementThreshold": 10,
                  "restSpeedThreshold": 10,
                  "stiffness": 1000,
                },
              },
            }
          }
        >
          <View
            pointerEvents="box-none"
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <View
              collapsable={false}
              forwardedRef={[Function]}
              onGestureHandlerEvent={[Function]}
              onGestureHandlerStateChange={[Function]}
              style={
                Object {
                  "flex": 1,
                  "transform": Array [
                    Object {
                      "translateX": 0,
                    },
                    Object {
                      "translateX": 0,
                    },
                  ],
                }
              }
            >
              <View
                pointerEvents="none"
                style={
                  Object {
                    "backgroundColor": "rgb(242, 242, 242)",
                    "bottom": 0,
                    "left": 0,
                    "position": "absolute",
                    "shadowColor": "#000",
                    "shadowOffset": Object {
                      "height": 1,
                      "width": -1,
                    },
                    "shadowOpacity": 0.3,
                    "shadowRadius": 5,
                    "top": 0,
                    "width": 3,
                  }
                }
              />
              <View
                style={
                  Array [
                    Object {
                      "flex": 1,
                      "overflow": "hidden",
                    },
                    Array [
                      Object {
                        "backgroundColor": "rgb(242, 242, 242)",
                      },
                      undefined,
                    ],
                  ]
                }
              >
                <View
                  style={
                    Object {
                      "flex": 1,
                      "flexDirection": "column-reverse",
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "flex": 1,
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "backgroundColor": "#1E1F28",
                          "flex": 1,
                          "paddingBottom": 20,
                          "paddingTop": 10,
                        }
                      }
                    >
                      <View
                        style={
                          Object {
                            "alignSelf": "center",
                            "flex": 1,
                            "justifyContent": "center",
                            "width": "90%",
                          }
                        }
                      >
                        <Text
                          style={
                            Object {
                              "color": "#ABB4BD",
                              "fontSize": 18,
                              "fontWeight": "bold",
                            }
                          }
                        >
                          Valor da transação(R$)
                        </Text>
                        <TextInput
                          allowFontScaling={true}
                          keyboardType="decimal-pad"
                          onChangeText={[Function]}
                          placeholder="Valor"
                          placeholderTextColor="#F5F5F5"
                          rejectResponderTermination={true}
                          style={
                            Object {
                              "color": "#F5F5F5",
                              "marginLeft": 5,
                              "padding": 15,
                            }
                          }
                          underlineColorAndroid="#F5F5F5"
                        />
                        <Text
                          style={
                            Object {
                              "color": "#ABB4BD",
                              "fontSize": 18,
                              "fontWeight": "bold",
                            }
                          }
                        >
                          Descrição
                        </Text>
                        <TextInput
                          allowFontScaling={true}
                          onChangeText={[Function]}
                          placeholder="Descrição"
                          placeholderTextColor="#F5F5F5"
                          rejectResponderTermination={true}
                          style={
                            Object {
                              "color": "#F5F5F5",
                              "marginLeft": 5,
                              "padding": 15,
                            }
                          }
                          underlineColorAndroid="#F5F5F5"
                        />
                        <Text
                          style={
                            Object {
                              "color": "#ABB4BD",
                              "fontSize": 18,
                              "fontWeight": "bold",
                            }
                          }
                        >
                          Data
                        </Text>
                        <View
                          accessible={true}
                          focusable={false}
                          onClick={[Function]}
                          onResponderGrant={[Function]}
                          onResponderMove={[Function]}
                          onResponderRelease={[Function]}
                          onResponderTerminate={[Function]}
                          onResponderTerminationRequest={[Function]}
                          onStartShouldSetResponder={[Function]}
                          style={
                            Object {
                              "opacity": 1,
                            }
                          }
                        >
                          <View
                            accessible={true}
                            focusable={true}
                            onClick={[Function]}
                            onResponderGrant={[Function]}
                            onResponderMove={[Function]}
                            onResponderRelease={[Function]}
                            onResponderTerminate={[Function]}
                            onResponderTerminationRequest={[Function]}
                            onStartShouldSetResponder={[Function]}
                            style={
                              Array [
                                Object {
                                  "width": 142,
                                },
                                Object {
                                  "width": 115,
                                },
                              ]
                            }
                          >
                            <View
                              style={
                                Array [
                                  Object {
                                    "alignItems": "center",
                                    "flexDirection": "row",
                                    "height": 40,
                                    "justifyContent": "center",
                                  },
                                  undefined,
                                ]
                              }
                            >
                              <View
                                style={
                                  Array [
                                    Object {
                                      "alignItems": "center",
                                      "borderColor": "#aaa",
                                      "borderWidth": 1,
                                      "flex": 1,
                                      "height": 40,
                                      "justifyContent": "center",
                                    },
                                    Object {
                                      "borderWidth": 0,
                                    },
                                    false,
                                    false,
                                  ]
                                }
                              >
                                <Text
                                  allowFontScaling={true}
                                  style={
                                    Array [
                                      Object {
                                        "color": "#333",
                                      },
                                      Object {
                                        "color": "#F5F5F5",
                                      },
                                    ]
                                  }
                                >
                                  09/08/2020
                                </Text>
                              </View>
                              <Modal
                                animationType="none"
                                hardwareAccelerated={false}
                                onRequestClose={[Function]}
                                supportedOrientations={
                                  Array [
                                    "portrait",
                                    "portrait-upside-down",
                                    "landscape",
                                    "landscape-left",
                                    "landscape-right",
                                  ]
                                }
                                transparent={true}
                                visible={false}
                              >
                                <View
                                  style={
                                    Object {
                                      "flex": 1,
                                    }
                                  }
                                >
                                  <View
                                    accessible={true}
                                    focusable={true}
                                    onClick={[Function]}
                                    onResponderGrant={[Function]}
                                    onResponderMove={[Function]}
                                    onResponderRelease={[Function]}
                                    onResponderTerminate={[Function]}
                                    onResponderTerminationRequest={[Function]}
                                    onStartShouldSetResponder={[Function]}
                                    style={
                                      Object {
                                        "alignItems": "flex-end",
                                        "backgroundColor": "#00000077",
                                        "flex": 1,
                                        "flexDirection": "row",
                                      }
                                    }
                                  >
                                    <View
                                      accessible={true}
                                      focusable={false}
                                      onClick={[Function]}
                                      onResponderGrant={[Function]}
                                      onResponderMove={[Function]}
                                      onResponderRelease={[Function]}
                                      onResponderTerminate={[Function]}
                                      onResponderTerminationRequest={[Function]}
                                      onStartShouldSetResponder={[Function]}
                                      style={
                                        Object {
                                          "flex": 1,
                                        }
                                      }
                                    >
                                      <View
                                        style={
                                          Object {
                                            "backgroundColor": "#fff",
                                            "height": 0,
                                            "overflow": "hidden",
                                          }
                                        }
                                      >
                                        <View
                                          pointerEvents="auto"
                                        >
                                          <View
                                            style={
                                              Array [
                                                Object {
                                                  "borderTopColor": "#ccc",
                                                  "borderTopWidth": 1,
                                                  "marginTop": 42,
                                                },
                                                undefined,
                                              ]
                                            }
                                          >
                                            <RCTDatePicker
                                              date={1596942000000}
                                              maximumDate={1609380000000}
                                              mode="date"
                                              onChange={[Function]}
                                              onResponderTerminationRequest={[Function]}
                                              onStartShouldSetResponder={[Function]}
                                              style={
                                                Object {
                                                  "height": 216,
                                                }
                                              }
                                              timeZoneOffsetInMinutes={null}
                                            />
                                          </View>
                                        </View>
                                        <View
                                          accessible={true}
                                          focusable={true}
                                          onClick={[Function]}
                                          onResponderGrant={[Function]}
                                          onResponderMove={[Function]}
                                          onResponderRelease={[Function]}
                                          onResponderTerminate={[Function]}
                                          onResponderTerminationRequest={[Function]}
                                          onStartShouldSetResponder={[Function]}
                                          style={
                                            Array [
                                              Object {
                                                "alignItems": "center",
                                                "flexDirection": "row",
                                                "height": 42,
                                                "justifyContent": "center",
                                                "paddingHorizontal": 20,
                                                "position": "absolute",
                                                "top": 0,
                                              },
                                              Object {
                                                "left": 0,
                                              },
                                              undefined,
                                            ]
                                          }
                                        >
                                          <Text
                                            allowFontScaling={true}
                                            style={
                                              Array [
                                                Object {
                                                  "color": "#46cf98",
                                                  "fontSize": 16,
                                                },
                                                Object {
                                                  "color": "#666",
                                                },
                                                undefined,
                                              ]
                                            }
                                          >
                                            Cancelar
                                          </Text>
                                        </View>
                                        <View
                                          accessible={true}
                                          focusable={true}
                                          onClick={[Function]}
                                          onResponderGrant={[Function]}
                                          onResponderMove={[Function]}
                                          onResponderRelease={[Function]}
                                          onResponderTerminate={[Function]}
                                          onResponderTerminationRequest={[Function]}
                                          onStartShouldSetResponder={[Function]}
                                          style={
                                            Array [
                                              Object {
                                                "alignItems": "center",
                                                "flexDirection": "row",
                                                "height": 42,
                                                "justifyContent": "center",
                                                "paddingHorizontal": 20,
                                                "position": "absolute",
                                                "top": 0,
                                              },
                                              Object {
                                                "right": 0,
                                              },
                                              undefined,
                                            ]
                                          }
                                        >
                                          <Text
                                            allowFontScaling={true}
                                            style={
                                              Array [
                                                Object {
                                                  "color": "#46cf98",
                                                  "fontSize": 16,
                                                },
                                                undefined,
                                              ]
                                            }
                                          >
                                            Confirmar
                                          </Text>
                                        </View>
                                      </View>
                                    </View>
                                  </View>
                                </View>
                              </Modal>
                            </View>
                          </View>
                        </View>
                      </View>
                      <View
                        accessible={true}
                        focusable={true}
                        onClick={[Function]}
                        onResponderGrant={[Function]}
                        onResponderMove={[Function]}
                        onResponderRelease={[Function]}
                        onResponderTerminate={[Function]}
                        onResponderTerminationRequest={[Function]}
                        onStartShouldSetResponder={[Function]}
                        style={
                          Object {
                            "alignItems": "center",
                            "alignSelf": "center",
                            "backgroundColor": "#EF3651",
                            "borderRadius": 25,
                            "justifyContent": "center",
                            "opacity": 1,
                            "padding": 15,
                            "width": "90%",
                          }
                        }
                      >
                        <Text
                          style={
                            Object {
                              "color": "#F5F5F5",
                              "fontSize": 16,
                              "fontWeight": "bold",
                            }
                          }
                        >
                          Adicionar Transação
                        </Text>
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RNCSafeAreaProvider>
</View>
`;
